dev/generate_protos.py

```diff
@@ -0,0 +1,203 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+cache_dir = "build/cache"
+temp_gencode_dir = "build/proto_gencode"
+
+mlflow_protos_dir = "mlflow/protos"
+test_protos_dir = "tests/protos"
+
+
+def gen_protos(proto_dir, proto_files, lang, protoc_bin, protoc_include_path, out_dir):
+    assert lang in ["python", "java"]
+
+    subprocess.check_call(
+        [
+            protoc_bin,
+            f"-I={protoc_include_path}",
+            f"-I={proto_dir}",
+            f"--{lang}_out={out_dir}",
+            *[f"{proto_dir}/{proto_file}" for proto_file in proto_files],
+        ]
+    )
+
+
+def apply_python_gencode_replacement(file_path):
+    content = file_path.read_text()
+
+    for old, new in python_gencode_replacements:
+        content = content.replace(old, new)
+
+    file_path.write_text(content, encoding="UTF-8")
```

Comment:
```suggestion
    file_path.write_text(content)
```

Reply:
I think explicitly setting encoding is safer to avoid encoding issue. Some OS might set default encoding to other format.

Reply:
do we run this on some OS that set default encoding to other format?

Reply:
I'd remove `encoding`, but fine to keep it since it's harmless.

Reply:
makes sense, didn't know that